target/linux/ramips/image/mt7621.mk

```diff
@@ -1500,6 +1500,16 @@ define Device/sercomm_na502s
 endef
 TARGET_DEVICES += sercomm_na502s
 
+define Device/snr_snr-cpe-me2-lite
+  $(Device/dsa-migration)
+  $(Device/uimage-lzma-loader)
+  IMAGE_SIZE := 16064k
+  DEVICE_VENDOR := SNR
+  DEVICE_MODEL := SNR-CPE-ME2-Lite
```

Comment:
Is it possible to make a factory.img image to install from the OEM firmware web interface?
```
IMAGES += factory.bin
  IMAGE/factory.bin := append-kernel | append-rootfs | check-size
```
Or are there any signatures from the manufacturer that need to be added?

Reply:
At this stage, this is not possible, the original firmware is checked for compliance with the CRC header, rootfs, the model name in the header, and a few more parameters.

Reply:
Is there a description of the header structure? After all, you can add `define Build/snr-factory`, for example, to generate a header.

Reply:
It seems like a normal kernel header, differing only in the image name and the last bytes
```
$ binwalk SNR-CPE-ME2-Lite-1.10.0.0811211051.bin

DECIMAL       HEXADECIMAL     DESCRIPTION
--------------------------------------------------------------------------------
0             0x0             uImage header, header size: 64 bytes, header CRC: 0xC2301DD7, created: 2021-11-08 05:56:26, image size: 12030303 bytes, Data Address: 0x81001000, Entry Point: 0x812E0220, data CRC: 0xC5C2175C, OS: Linux, CPU: MIPS, image type: OS Kernel Image, compression type: lzma, image name: "SNR-CPE-ME2-Lite"
64            0x40            LZMA compressed data, properties: 0x6D, dictionary size: 134217728 bytes, uncompressed size: 3948992 bytes
1336855       0x146617        Squashfs filesystem, little endian, version 4.0, compression:xz, size: 10693512 bytes, 1297 inodes, blocksize: 131072 bytes, created: 2021-11-08 05:56:24

$ hexdump -Cn $((0x40)) SNR-CPE-ME2-Lite-1.10.0.0811211051.bin
00000000  27 05 19 56 c2 30 1d d7  61 88 bc 0a 00 b7 91 5f  |'..V.0..a......_|
00000010  81 00 10 00 81 2e 02 20  c5 c2 17 5c 05 05 02 03  |....... ...\....|
00000020  53 4e 52 2d 43 50 45 2d  4d 45 32 2d 4c 69 74 65  |SNR-CPE-ME2-Lite|
00000030  00 00 00 00 00 00 00 00  00 00 00 00 00 14 66 17  |..............f.|
00000040
```

Reply:
Added UIMAGE_NAME, now you can update via the web from the original firmware (version 1.10.4 and later).

Reply:
Shouldn't it be necessary to add:
```
  IMAGES += factory.bin
  IMAGE/factory.bin := append-kernel | append-rootfs | check-size
```
To create an install image from the manufacturer's firmware web interface?

Reply:
Updating to openwrt via the web interface is possible starting from manufacturer's firmware version 1.10.4 or later, sysupgrade.bin is suitable.

```
  IMAGES += factory.bin
  IMAGE/factory.bin := append-kernel | append-rootfs | check size

```
The firmware generated in this way will not pass the crc check.